### PR TITLE
Feature/te 294 e4 rcp

### PR DESCRIPTION
**Pull request by gunther-bachmann**
especially in folder rcp4 are several experimental codings which will have to be cleaned up with the ticket to create aml mask/step selection dialog for drag and drop.

windows line feeds that crept into the repo will have to be incrementally cleaned up, sorry about that

### CODE REVIEW

File: rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/EventBrokerUser.xtend [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.eclipse.e4.core.di.annotations.Optional
+import org.eclipse.e4.core.di.extensions.EventTopic
+import org.eclipse.e4.core.services.events.IEventBroker
+
+
+/** example usage of the event broker in e4 */
+class EventBrokerUser{
+	
+	new(){
+		System.out.println("EventBrokerUser is created")
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/EventBrokerUser.xtend]: We should use a logger here.

File: rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/ApplicationWorkbenchWindowAdvisor.xtend [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+	override postWindowOpen() {
+		removeUnwantedWizards
+	}
+
+	def removeUnwantedWizards() {
+		val wizardRegistry = PlatformUI.workbench.newWizardRegistry
+		val categories = PlatformUI.workbench.newWizardRegistry.rootCategory.categories
+		categories.allWizards.filter[!(category.id.matches("org.eclipse.ui.Basic") || id.matches("org.testeditor.*"))].
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/ApplicationWorkbenchWindowAdvisor.xtend]: Could you assign the filter result to a variable first and then proceed?

File: rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/ApplicationWorkbenchWindowAdvisor.xtend [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			configureEditorAreaDropListener(new EditorAreaDropAdapter(PlatformUI.workbench.activeWorkbenchWindow)) 
+		]
+	}
+
+
+	override postWindowOpen() {
+		removeUnwantedWizards
+	}
+
+	def removeUnwantedWizards() {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/ApplicationWorkbenchWindowAdvisor.xtend]: I like return types for public methods :)

File: tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/AbstractTest.xtend [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import org.eclipse.xtext.junit4.XtextRunner
+import org.eclipse.xtext.junit4.validation.ValidationTestHelper
+import org.eclipse.xtext.util.Modules2
+import org.junit.runner.RunWith
+import org.testeditor.dsl.common.testing.AssertionHelper
+import org.testeditor.tcl.dsl.TclRuntimeModule
+import org.testeditor.tcl.dsl.TclStandaloneSetup
+
+@RunWith(XtextRunner)
+abstract class AbstractTest extends AbstractXtextTests {
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/AbstractTest.xtend]: Don't we have an AbstractTest in TCL tests already? The one from the dsl bundle.

File: rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/NewWizard.xtend [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+	override performFinish() {
+		val result = super.performFinish()
+
+		val folder = newProject.getFolder("src-gen");
+		folder.create(false, true, null)
+
+		newProject.addNature(JavaCore.NATURE_ID)
+		JavaCore.create(newProject) =>
+			[
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/NewWizard.xtend]: should be reformatted, the `[` should be on the same line as the `=>`

File: rcp/org.testeditor.rcp4.views/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ org.eclipse.emf.ecore.presentation,
+ org.eclipse.emf.ecore.provider,
+ org.eclipse.emf.ecore.resource,
+ org.eclipse.emf.ecore.resource.impl,
+ org.eclipse.emf.ecore.util,
+ org.eclipse.emf.ecore.xcore,
+ org.eclipse.emf.ecore.xcore.exporter,
+ org.eclipse.emf.ecore.xcore.exporter.ui,
+ org.eclipse.emf.ecore.xcore.formatting,
+ org.eclipse.emf.ecore.xcore.generator,
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views/META-INF/MANIFEST.MF]: For my taste those are way to many imports. Why not require the bundle instead?

File: aml/org.testeditor.aml.dsl/META-INF/MANIFEST.MF [hunk 7]
```diff
@@ -9,6 +9,9 @@ Require-Bundle: org.eclipse.xtext,
  org.eclipse.equinox.common;bundle-version="3.7.0",
  org.eclipse.emf.ecore.xcore,
  org.testeditor.aml.model,
+ org.eclipse.osgi,
+ org.eclipse.xtext.generator,
```
Comment (franzbecker) [inline aml/org.testeditor.aml.dsl/META-INF/MANIFEST.MF]: We should not need to depend on the generator here, should we?
This should go into the `Automated Management of Dependencies` part.

File: aml/org.testeditor.aml.model/META-INF/MANIFEST.MF [hunk 8]
```diff
@@ -10,7 +10,8 @@ Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Export-Package: org.testeditor.aml,
  org.testeditor.aml.impl,
  org.testeditor.aml.util
-Require-Bundle: org.eclipse.core.runtime,
+Require-Bundle: org.eclipse.emf.ecore.xcore,
```
Comment (franzbecker) [inline aml/org.testeditor.aml.model/META-INF/MANIFEST.MF]: Why do we need the dependency to `org.eclipse.emf.ecore.xcore` ?

File: rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/NewWizard.xtend (hunk 5 above)
Comment (gunther-bachmann) [inline rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/NewWizard.xtend]: autoformatting does that

File: tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/AbstractTest.xtend (hunk 4 above)
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/AbstractTest.xtend]: yeah, its already in org.testeditor.tcl.dsl.tests but neither test is a plugin project. the question is then how do I import the existing class...(e.g. make the tests plugin projects too)

File: rcp/org.testeditor.rcp4.views/META-INF/MANIFEST.MF (hunk 6 above)
Comment (gunther-bachmann) [inline rcp/org.testeditor.rcp4.views/META-INF/MANIFEST.MF]: result of experimenting with lots of different apis. will have to cleanup those ...

File: rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/ApplicationWorkbenchWindowAdvisor.xtend (hunk 2 above)
Comment (gunther-bachmann) [inline rcp/org.testeditor.rcp4.product/src/org/testeditor/rcp4/product/ApplicationWorkbenchWindowAdvisor.xtend]: elaborate

**Pull request merged**